dag-cbor/src/decode.rs

```diff
@@ -411,156 +431,160 @@ impl<T: TryReadCbor> TryReadCbor for BTreeMap<String, T> {
 }
 impl_decode!(BTreeMap<String, T>);
 
-impl TryReadCbor for Ipld {
+impl<C, H> TryReadCbor for Ipld<C, H>
+where
+    C: Into<u64> + TryFrom<u64> + Copy,
+    H: Into<u64> + TryFrom<u64> + Copy,
+{
     fn try_read_cbor<R: Read>(r: &mut R, major: u8) -> Result<Option<Self>> {
         let ipld = match major {
             // Major type 0: an unsigned integer
-            0x00..=0x17 => Ipld::Integer(major as i128),
-            0x18 => Ipld::Integer(read_u8(r)? as i128),
-            0x19 => Ipld::Integer(read_u16(r)? as i128),
-            0x1a => Ipld::Integer(read_u32(r)? as i128),
-            0x1b => Ipld::Integer(read_u64(r)? as i128),
+            0x00..=0x17 => Ipld::<C, H>::Integer(major as i128),
```

Comment:
Changing it to `Self` is probably better

Reply:
That's waaaay better.